app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,33 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def post_params
+    params
+      .require(:post)
+      .permit(:title, :text)
+      .merge(author: current_user, comments_counter: 0, likes_counter: 0)
+  end
```

Comment:
- [Optional] To prevent this method from being accessed outside the controller, it's always made private. By convention, this method should not be called from any external object.

Reply:
@FrederickMih Thank you for your valuable suggestion. I have implemented the suggested changes https://github.com/NayanKaran/blog-using-rails/pull/8/commits/6c8b24eeef6e940e403004f42122e500d18023c7 .